command/output_test.go

```diff
@@ -24,23 +23,25 @@ func TestOutput(t *testing.T) {
 
 	statePath := testStateFile(t, originalState)
 
-	ui := new(cli.MockUi)
+	view, done := testView(t)
```

Comment:
It's a shame that the setup here doesn't make it easy to pass in a higher-level mock while testing this layer, so these tests can focus on just the controller and not also implicitly testing one of the views, but I must admit I don't see a clean way of doing it that doesn't either make things much more complex or end up skipping testing the overall command, so this does seem like a good compromise for now. Perhaps as we implement a few more of these we'll start to see some patterns that suggest ways to isolate the controller and the view differently, but this is still a great improvement over what we had before!

Reply:
I agree with this being unfortunate. I have an idea for how we might work around this (moving the view constructor to the `views.View` struct, making `views.View` an interface, and allowing injection of a call-logging mock as the `testView`) but it's again not something I want to add this PR.

I would also add that several of the output command tests ought to be moved to the views package, as the controller code coverage is already good enough. Again, I am trying to minimize the diff (although looking at it that might not be obvious) so I want to leave that for a follow-up.